tests/phpunit/CRM/Contact/Import/Form/MapFieldTest.php

```diff
@@ -362,6 +368,16 @@ public function getHeaderMatchDataProvider() {
     ];
   }
 
+  /**
+   * This is accessed by virtue of the MetaDataTrait being included.
+   *
+   * The use of the metadataTrait came from a transitional refactor
+   * but it probably should be phased out again.
+   */
+  protected function getContactType() {
+    return $this->_contactType ?? 'Individual';
```

Comment:
This is a bit misleading because `$this` is the test class not the form so the function always returns Individual, and it just happens to work out because the tests involved are testing Individual. So it's not that it should "probably" be phased out again, it does eventually need to be "fixed".

Reply:
@demeritcowboy yeah - there was a reason for the trait when it was added (ie the code was so confusing that there needed to be a way to extract & clean up some parts)  - but I think there are some better way to achieve the same things - hence it's the trait I want to phase out I did write some notes about the restructure & the goal https://lab.civicrm.org/dev/core/-/issues/3414  - but in terms of the trait I think the functions probably eventually belong to the `Parser` class - it's just that it has traditionally been so painful to call that class.

I've started to move towards things like this though https://github.com/civicrm/civicrm-core/pull/23288/files#diff-6c1a95e4ed243dbed9e0b7c980cd2fed4e538c09863b6ae45a2e0538d1db8321R668 (which is an example of accessing something from the MetadataTrait without `using` it